Cesium.CodeGen/Ir/Expressions/FunctionCallExpression.cs

```diff
@@ -39,36 +39,78 @@ public IExpression Lower(IDeclarationScope scope)
         var functionName = _function.Identifier;
         var callee = scope.Functions.GetValueOrDefault(functionName)
                      ?? throw new CompilationException($"Function \"{functionName}\" was not found.");
+        int firstVarArgArgument = 0;
+        if (callee.Parameters is { } parameters)
+        {
+            if (parameters.IsVarArg)
+            {
+                firstVarArgArgument = parameters.Parameters.Count;
+            }
+        }
+
         return new FunctionCallExpression(
             _function,
             callee,
-            _arguments.Select(a => a.Lower(scope)).ToList());
+            _arguments.Select((a, index) =>
+            {
+                if (index >= firstVarArgArgument)
+                {
+                    var expressionType = a.GetExpressionType(scope);
+                    if (expressionType.Equals(scope.CTypeSystem.Float))
+                    {
+                        // Seems to be float always use float-point registers and as such we need to covert to double.
+                        return new TypeCastExpression(scope.CTypeSystem.Double, a.Lower(scope));
+                    }
+                    else
+                    {
+                        return a.Lower(scope);
+                    }
+                }
+
+                return a.Lower(scope);
+            }).ToList());
     }
 
     public void EmitTo(IEmitScope scope)
     {
+        VariableDefinition? varArgBuffer = null;
         var explicitParametersCount = _callee!.Parameters?.Parameters.Count ?? 0;
+        var varArgParametersCount = _arguments.Count - explicitParametersCount;
+        if (_callee!.Parameters?.IsVarArg == true)
+        {
+            // Using sparse population of the parameters on the stack. 8 bytes should be enough for anybody.
+            // Also we need perform localloc on empty stack, so we will use local variable to save vararg buffer to temporary variable.
+            if (varArgParametersCount == 0)
+            {
+                scope.AddInstruction(OpCodes.Ldc_I4_0);
```

Comment:
What's the purpose of this `Ldc_I4_0`?

Reply:
This is null.

Reply:
Basically I borrow that from Sharplab. Ldnull works fine too and can change to it if you prefer, but seems to be both these methods are okay

Reply:
I would prefer `ldnull`. Less questions that way.

Or, if this is some sort of optimization (say, `ldc_i4_0` takes less space in byte code), then let's have a helper method that chooses an appropriate form of `ldnull`-compatible instruction.

Reply:
Done. Anything else? Size is the same since no parameters passed to opcode